src/components/Image/index.native.tsx

```diff
@@ -1,35 +1,31 @@
-import lodashGet from 'lodash/get';
 import React from 'react';
 import RNFastImage from 'react-native-fast-image';
 import {withOnyx} from 'react-native-onyx';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import {defaultProps, imagePropTypes} from './imagePropTypes';
 import RESIZE_MODES from './resizeModes';
+import {DimensionsCacheValue, FastImageSource, ImageOnyxProps, ImagePropsWithOnyx} from './types';
 
-const dimensionsCache = new Map();
+const dimensionsCache = new Map<string, DimensionsCacheValue>();
 
-function resolveDimensions(key) {
+function resolveDimensions(key: string): DimensionsCacheValue | undefined {
     return dimensionsCache.get(key);
 }
 
-function Image(props) {
-    // eslint-disable-next-line react/destructuring-assignment
-    const {source, isAuthTokenRequired, session, ...rest} = props;
-
-    let imageSource = source;
-    if (source && source.uri && typeof source.uri === 'number') {
+function Image({source, isAuthTokenRequired, session, ...rest}: ImagePropsWithOnyx) {
+    let imageSource: FastImageSource = source;
+    if (typeof source !== 'number' && typeof source.uri === 'number') {
         imageSource = source.uri;
     }
-    if (typeof imageSource !== 'number' && isAuthTokenRequired) {
-        const authToken = lodashGet(props, 'session.encryptedAuthToken', null);
+    if (typeof source !== 'number' && isAuthTokenRequired) {
```

Comment:
Same, it was using `imageSource` and you changed to `source`. Why?

Reply:
Right, we have to check for both, as `source` can be a number and therefore it can't be spread.